File: src/codex_team/paths.py
"""XDG-aware path resolution helpers."""

from __future__ import annotations

import os
import tempfile
from pathlib import Path

_APP = "codex-team"


def _home() -> Path:
    return Path(os.environ.get("HOME", "/"))


def xdg_config_dir() -> Path:
    base = os.environ.get("XDG_CONFIG_HOME") or str(_home() / ".config")
    return Path(base) / _APP


def xdg_data_dir() -> Path:
    base = os.environ.get("XDG_DATA_HOME") or str(_home() / ".local/share")
    return Path(base) / _APP


def xdg_runtime_dir() -> Path:
    base = os.environ.get("XDG_RUNTIME_DIR")
    if not base:
        base = os.environ.get("TMPDIR") or tempfile.gettempdir()
    return Path(base) / _APP


def default_socket_path() -> Path:
    return xdg_runtime_dir() / "daemon.sock"


def default_pid_path() -> Path:
    return xdg_data_dir() / "daemon.pid"


def default_log_path() -> Path:
    return xdg_data_dir() / "daemon.log"


def default_registry_path() -> Path:
    return xdg_data_dir() / "registry.json"


def session_dir(data_dir: Path, name: str) -> Path:
    return data_dir / "sessions" / name

File: plugins/codex-team/src/platform/paths.ts
import os from "node:os";
import path from "node:path";

import { isWindows } from "./os";

const APP = "codex-team";

function homeDir(): string {
  return process.env.HOME || os.homedir() || (isWindows ? process.env.USERPROFILE || "C:\\" : "/");
}

export function resolveConfigDir(): string {
  if (process.env.CODEX_TEAM_CONFIG_DIR) {
    return path.resolve(process.env.CODEX_TEAM_CONFIG_DIR);
  }
  if (isWindows) {
    return path.join(process.env.APPDATA || path.join(homeDir(), "AppData", "Roaming"), APP);
  }
  return path.join(process.env.XDG_CONFIG_HOME || path.join(homeDir(), ".config"), APP);
}

export function resolveDataDir(configured = ""): string {
  if (configured) {
    return path.resolve(configured);
  }
  if (process.env.CODEX_TEAM_DAEMON_DATA_DIR) {
    return path.resolve(process.env.CODEX_TEAM_DAEMON_DATA_DIR);
  }
  if (
    process.env.CLAUDE_PLUGIN_DATA &&
    (process.env.CLAUDE_PLUGIN_ROOT || process.env.CLAUDE_PLUGIN_DATA.includes(APP))
  ) {
    return path.join(path.resolve(process.env.CLAUDE_PLUGIN_DATA), "data");
  }
  if (isWindows) {
    return path.join(process.env.LOCALAPPDATA || path.join(homeDir(), "AppData", "Local"), APP);
  }
  return path.join(process.env.XDG_DATA_HOME || path.join(homeDir(), ".local", "share"), APP);
}

export function resolveRuntimeDir(configuredDataDir = ""): string | null {
  if (isWindows) {
    return null;
  }
  if (process.env.CODEX_TEAM_DAEMON_RUNTIME_DIR) {
    return path.resolve(process.env.CODEX_TEAM_DAEMON_RUNTIME_DIR);
  }
  if (process.env.CLAUDE_PLUGIN_DATA) {
    return path.join(path.resolve(process.env.CLAUDE_PLUGIN_DATA), "runtime");
  }
  void configuredDataDir;
  return path.join(process.env.XDG_RUNTIME_DIR || process.env.TMPDIR || os.tmpdir(), APP);
}

export function resolvePidPath(dataDir: string, configured = ""): string {
  return path.resolve(configured || process.env.CODEX_TEAM_DAEMON_PID_PATH || path.join(dataDir, "daemon.pid"));
}

export function resolveLogPath(dataDir: string): string {
  return path.join(dataDir, "daemon.log");
}

export function resolveRegistryPath(dataDir: string): string {
  return path.join(dataDir, "registry.json");
}

export function sessionDir(dataDir: string, workspace: string, name: string): string {
  return path.join(dataDir, "sessions", workspace, name);
}

export function clientsDir(dataDir: string): string {
  return path.join(dataDir, "clients");
}

export function alarmsDir(dataDir: string): string {
  return path.join(dataDir, "alarms");
}

export function workspaceAlarmsDir(dataDir: string, workspace: string): string {
  return path.join(alarmsDir(dataDir), workspace);
}

export function workspaceEnvFile(projectDir: string): string {
  return path.join(projectDir, ".codex-team", "workspace.env");
}

export function clientEnvFile(projectDir: string): string {
  return path.join(projectDir, ".codex-team", "client.env");
}
